test/contrib/test_pyopenssl_dependencies.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+import unittest
+
+from nose.plugins.skip import SkipTest
+
+try:
+    from urllib3.contrib.pyopenssl import inject_into_urllib3
+except ImportError as e:
+    raise SkipTest('Could not import PyOpenSSL: %r' % e)
+
+from mock import patch
+
+class TestPyOpenSSLInjection(unittest.TestCase):
+    """
+    Tests for error handling in pyopenssl's 'inject_into urllib3'
+    """
+    def test_inject_validate_fail(self):
+        """
+        Injection should not be supported if we are missing required dependencies.
+        """
+        with patch("cryptography.x509.extensions.Extensions") as mock:
+            del mock.get_extension_for_class
+            self.assertRaises(ImportError, inject_into_urllib3)
```

Comment:
Yeah -- "can we call `inject_into_urllib3` more than once" was the issue that you raised prior to this one :-)  Anyway, I'll make the try/finally change.  (Upon further thought, I think I can clean it up a bit and skip the `except` clause.)